sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/models/StorageAccount.java

```diff
@@ -746,6 +814,51 @@ interface WithEncryption {
+             * If the provided user-assigned identity is null, this method is equivalent to {@link WithEncryption#withEncryptionKeyFromKeyVault(String, String, String)},
+             * meaning system-assigned identity will be used to access the KeyVault.
+             *
+             * This requires managed service identity on storage account
+             * (via {@link WithManagedServiceIdentity#withSystemAssignedManagedServiceIdentity()} or
+             * {@link DefinitionStages.WithUserAssignedManagedServiceIdentity#withExistingUserAssignedManagedServiceIdentity(Identity)}),
+             * and GET, WRAP_KEY, UNWRAP_KEY access policy on key vault for the managed service identity.
+             *
+             * @param keyVaultUri the uri to KeyVault
+             * @param keyName the KeyVault key name
+             * @param keyVersion the KeyVault key version
+             * @param userAssignedIdentity user-assigned identity to access the KeyVault
+             * @return the next stage of storage account update
+             */
+            Update withEncryptionKeyFromKeyVault(String keyVaultUri, String keyName, String keyVersion, Identity userAssignedIdentity);
+
+            /**
+             * Specifies the KeyVault key to be used as key for encryption and the user-assigned identity to access the KeyVault,
+             * meaning this Storage Account will be encrypted with Customer-managed keys.
+             * If the provided user-assigned identity is null, this method is equivalent to {@link WithEncryption#withEncryptionKeyFromKeyVault(String, String, String)},
+             * meaning system-assigned identity will be used to access the KeyVault.
+             *
+             * This requires managed service identity on storage account
+             * (via {@link WithManagedServiceIdentity#withSystemAssignedManagedServiceIdentity()} or
+             * {@link DefinitionStages.WithUserAssignedManagedServiceIdentity#withExistingUserAssignedManagedServiceIdentity(Identity)}),
+             * and GET, WRAP_KEY, UNWRAP_KEY access policy on key vault for the managed service identity.
+             *
+             * @param keyVaultUri the uri to KeyVault
+             * @param keyName the KeyVault key name
+             * @param keyVersion the KeyVault key version
+             * @param userAssignedIdentityId ID of the user-assigned identity to access the KeyVault
+             * @return the next stage of storage account update
+             */
+            Update withEncryptionKeyFromKeyVault(String keyVaultUri, String keyName, String keyVersion, String userAssignedIdentityId);
+
+            /**
+             * Specifies the Microsoft-managed key to be used as key for encryption. This is the default encryption type.
+             *
+             * @return the next stage of storage account update
+             */
+            Update withEncryptionKeyFromStorage();
```

Comment:
A bit unsure about the name. candidate: "withEncryptionKeyFromMicrosoft" "withMicrosoftManagedEncryptionKey"

Reply:
I'm not sure either. We had a method `encryptionKeySource()` exposed, with two possible values `Microsoft.Storage` and `Microsoft.Keyvault`.

Current relations are:
* `withEncryptionKeyFromKeyVault` -> `Microsoft.Keyvault`
* `withEncryptionKeyFromStorage`(`withMicrosoftManagedEncryptionKey`) -> `Microsoft.Storage`

Guess `withMicrosoftManagedEncryptionKey` should be fine?

Reply:
Yeah, personally I am on `withMicrosoftManagedEncryptionKey`

@haolingdong-msft  your opinion as well?

Reply:
@haolingdong-msft 

`withEncryptionKeyFromKeyVault` -> encryption using Customer Managed Keys
`withEncryptionKeyFromStorage(withMicrosoftManagedEncryptionKey)` -> encryption using Microsoft Managed Keys

doc: [withEncryptionKeyFromStorage(withMicrosoftManagedEncryptionKey)](https://learn.microsoft.com/en-us/azure/storage/common/customer-managed-keys-overview)

Reply:
My gust feeling is the signature would be better to be: `withEncryptionKey(EncryptionKeyKindEnum)`. But as we already exposed `withEncryptionKeyFromKeyVault()`, I think it's impossible to change to`withEncryptionKey(EncryptionKeyKindEnum)`.

I don't have strong prereference for this. Maybe `withMicrosoftManagedEncryptionKey` to better align with the document.

Reply:
`EncryptionKeyKindEnum` probably won't work, as MMK and CMK be very different on required configs.